ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -249,6 +252,93 @@ object Queries {
     }
   }
 
+  /** Make the smallest number of queries from `queries` that still indicates
+    * which query or queries produced each contract.
+    *
+    * A contract cannot be produced more than once from a given resulting query,
+    * but may be produced more than once from different queries.  In each case, the
+    * `templateId` of the resulting [[DBContract]] is actually the 0-based index
```

Comment:
Since the index is in a sense "templateId +" with reference to argument data, it seems unfortunate to introduce another field to model a similar idea, when there is no semantic reason.

My perception of this choice is that we can have

1. a name that is occasionally a little inaccurate, but an aid to "intuition", for what it's worth; or
2. a name that reflects the uninterpreted, formal semantics of the field, but offers no "intuitive" aid to the reader.

I think you can guess which choice I generally favor, though I have chosen the other in this case due to the stronger pull of change conservatism. Which do you think is most valuable to us as maintainers of the code?

Reply:
We do have a type change, because `templateId` is type-parameterized. 😕

Reply:
oh perfect, apologies for missing that. In that case, I don’t really care about the name so for now I’d just keep it.